### PR TITLE
wanted to use URLs

### PR DESCRIPTION
**Pull request by zbenmo**
Played with Quick Draw - Mona Lisa, repeated your Eiffel experiment. - Wanted to use URLs directly to the web server as you have suggested. I have not used Prodigy.

Really interesting clusters with the Mona Lisa DS.

quick draw Mona Lisa

WKR,
Oren

### CODE REVIEW

**Comment by koaning** [comment]
In the future, please open an issue first to discuss the feature. That way we can agree on big-picture implementation stuff before putting any work in. 

That said ... interesting! Have you tried running this locally? While I don't mind adding this feature, part of me does worry that it could be slow. Partially because local images are always faster, but also because the current implementation doesn't use async. Could you comment on the performance?

**Comment by zbenmo** [comment]
Hi,

It was reasonable (a minute or a few) for ~3000 quick-draw images.

Mostly wanted to brag about me trying your package and finding interesting
pattern in the Mona Lisa dataset. Will not be offended if feature not
merged.


WKR,
Oren

**Comment by koaning** [comment]
I actually do really like the idea and it's super cool to see that you've gone ahead and replicated the video! I'm definately interested in the PR, it's just that I want to make sure it's not a performance hog before I recommend people to use this feature. 

Could you add a GIF of what the experience is like on your end? I might also be able to replicate it myself but I might not have time for that this week. 

On a personal note; you've found a really cool set of clusters here that's totally worth a blog-post. If you haven't started a blog yet; this would make a cool first entry.

**Comment by koaning** [comment]
If you're not planning to blog about it, would you mind if I tweet/toot about this? I _really_ like this example. 

Honestly, this is top 3 favorite lifetime PRs material right here.

**Comment by zbenmo** [comment]
Okay, so now I have an incentive to blog it. I will and share.

**Comment by koaning** [comment]
If you're looking for software; I heard great things about `hugo` but my own https://koaning.io blog is just a script that generates a blog via `mkdocs`. I highly recommend blogging in a TIL-manner. Blogposts don't have to be long and they can also just be a nice personal archive of thoughts you want to be able to remember/share. 

Here's an example from my blog: https://koaning.io/til/

You may also gain inspiration from Simon Willision: https://til.simonwillison.net/

**Comment by zbenmo** [comment]
Thanks.

https://zbenmo.medium.com/bulk-labelling-the-mona-lisa-98acd5ea8ce2

I will look into hugo in the future.


WKR,
Oren

File: bulk/cli/image.py [hunk 1]
```diff
@@ -17,10 +17,17 @@
...
 def encode_image(path):
-    with open(path, "rb") as image_file:
-        enc_str = base64.b64encode(image_file.read()).decode("utf-8")
+    if type(path) == str and path.startswith('http'):
+        http = urllib3.PoolManager()
+        contents = http.request('GET', path)
+        enc_str = base64.b64encode(contents.data).decode("utf-8")
+    else:
+        with open(path, "rb") as image_file:
+            enc_str = base64.b64encode(image_file.read()).decode("utf-8")
```
Comment (koaning) [inline bulk/cli/image.py:26]: Wouldn't it be a lot less computationally intensive to have the frontend do this fetching? Something like:

```python
if path.startswith('http'):
     return f'<img style="object-fit: scale-down;" width="100%" height="100%" src="{path}">'
```

**Review by koaning — CHANGES_REQUESTED** [review]
I think this can be a whole lot quicker, but lets discuss!

**Review by koaning — APPROVED** [review]
LGTM!

**Pull request merged**